src/session/session_prefetch.c

```diff
@@ -10,34 +10,26 @@
 
 /*
  * __wt_session_prefetch_check --
- *     Check to see whether cursors owned by this session might benefit from doing pre-fetch
- *
- * FIXME-WT-11758 Change this function so that when pre-fetching is enabled on the session level,
- *     all cursors associated with that session with automatically perform pre-fetching. Use the
- *     session-level configuration once it is introduced.
+ *     Check if pre-fetching work should be performed for a given ref.
  */
 bool
 __wt_session_prefetch_check(WT_SESSION_IMPL *session, WT_REF *ref)
 {
-    /* Internal threads should not be configured to do pre-fetching. */
-    if (!S2C(session)->prefetch_auto_on || F_ISSET(session, WT_SESSION_INTERNAL))
-        return (false);
-
     if (S2C(session)->prefetch_queue_count > WT_MAX_PREFETCH_QUEUE)
         return (false);
 
-    /*
-     * Don't deal with internal pages at the moment - finding the right content to preload based on
-     * internal pages is hard.
-     */
-    if (F_ISSET(ref, WT_REF_FLAG_INTERNAL))
-        return (false);
-
     if (session->pf.prefetch_disk_read_count == 1)
         WT_STAT_CONN_INCR(session, block_prefetch_disk_one);
 
-    /* A single read from disk is common - don't use it to guide pre-fetch behavior. */
-    if (session->pf.prefetch_disk_read_count < 2) {
+    /*
+     * Check if pre-fetching is enabled for this particular session. We don't perform pre-fetching
+     * on internal threads or internal pages (finding the right content to preload based on internal
+     * pages is hard), so check for that too. We also want to pre-fetch pages that have had at least
```

Comment:
It isn't "pages that have had at least one read", it is "sessions that have read at least one page from disk"